numpy/lib/function_base.py

```diff
@@ -5456,13 +5458,17 @@ def append(arr, values, axis=None):
     dimension(s)
 
     """
-    arr = asanyarray(arr)
+    if dtype is not None:
+        arr = asanyarray(arr, dtype=dtype)
+    else:
+        arr = asanyarray(arr)
+
     if axis is None:
         if arr.ndim != 1:
             arr = arr.ravel()
         values = ravel(values)
         axis = arr.ndim-1
```

Comment:
We could delete all of this code I think.  `concatenate` supports `axis=None` in exactly this way already?  I am wondering why it is there, but it may well out-date NumPy itself...

Mainly, casting `arr` to `dtype` ahead of time is also unnecessary.

Reply:
@seberg I delete the checking dtype if ```None``` code. I also found it unnecessary and only depended on passing the ```dtype``` to the ```concatenate``` function.